.github/workflows/test.yaml

```diff
@@ -261,6 +261,55 @@ jobs:
         fail_ci_if_error: true
 
 
+  fips-tests:
+    # Defined the host in which the container is executed
+    runs-on: ubuntu-latest
+    # Define container inside which the steps are executed
+    container:
+      image: photon:5.0
+      env:
+        NODE_ENV: development
+      options: --cpus 2
+    # Steps as normal.
+    steps:
+    - name: Setup the container
+      run: |
+        tdnf remove -y toybox
+        tdnf update -y
+        tdnf install -y sed
+        tdnf install -y coreutils \
+                openssl-fips-provider \
+                python3-pip \
+                python3-pip \
+                python3 \
+                python3-devel \
+                build-essential \
+                git \
+                shadow \
+                tzdata
+
+        pip3 install tox
+
+    - uses: actions/checkout@v3
+      with:
+        path: twisted
+
+    - name: Prepare for testenv
+      run: |
+        # run tests from user context
+        # for example, test_checker.py tests fail if run as root
+        useradd -m test
+        cp -r twisted /home/test
+        chown -R test /home/test
+
+    - name: Run tests
+      run: |
+        # don't hog the host's resources while testing
+        ulimit -S -n 1024
+        ulimit -H -n 1024
+        ulimit -a
+        su - test -c "cd twisted && tox -e alldeps-nocov-posix"
```

Comment:
the `-nocov-` is only as a start.

It would be nice to have the coverage reported

```
    - name: Prepare coverage
      if: ${{ !cancelled() }}
      run: |
        # sub-process coverage are generated in separate files so we combine them
        # to get an unified coverage for the local run.
        # The XML is generate to be used with 3rd party tools like diff-cover.
        python -m coverage combine
        python -m coverage xml -o coverage.xml -i
        python -m coverage report --skip-covered

    - uses: codecov/codecov-action@v3
      if: ${{ !cancelled() }}
      with:
        files: coverage.xml
        name: fips
        fail_ci_if_error: true
```